src/application-manager.coffee

```diff
@@ -0,0 +1,875 @@
+			.then (service) ->
+				res.status(200).json({ containerId: service.containerId })
+			.catch (err) ->
+				res.status(503).send(err?.message or err or 'Unknown error')
+
+		@router.get '/v1/apps/:appId', (req, res) =>
+			appId = checkString(req.params.appId)
+			@eventTracker.track('GET app (v1)', appId)
+			if !appId?
+				return res.status(400).send('Missing app id')
+			@applications.getCurrentApp(appId)
+			.then (app) ->
+				service = app?.services?[0]
+				return res.status(400).send('App not found') if !service?
+				return res.status(400).send('v1 endpoints are only allowed on single-container apps') if app.services.length > 1
+				# Don't return data that will be of no use to the user
+				appToSend = {
+					appId
+					containerId: service.containerId
+					env: _.omit(service.environment, constants.privateAppEnvVars)
+					commit: service.commit
+					releaseId: app.releaseId
+					imageId: service.image
+				}
+				res.json(appToSend)
+			.catch (err) ->
+				res.status(503).send(err?.message or err or 'Unknown error')
+
+		@router.post '/v1/purge', (req, res) =>
+			appId = checkString(req.body.appId)
+			force = checkTruthy(req.body.force)
+			if !appId?
+				errMsg = "App not found: an app needs to be installed for purge to work.
+						If you've recently moved this device from another app,
+						please push an app and wait for it to be installed first."
+				return res.status(400).send(errMsg)
+			Promise.using updateLock.lock(appId, { force }), =>
+				@applications.getCurrentApp(appId)
+				.then (app) =>
+					service = app?.services?[0]
+					return res.status(400).send('App not found') if !service?
+					return res.status(400).send('v1 endpoints are only allowed on single-container apps') if app.services.length > 1
+					@applications.executeStepAction(serviceAction('kill', service.serviceId, service, null, skipLock: true), { force })
+					.then =>
+						@applications.executeStepAction({
+							action: 'purge'
+							appId: app.appId
+							options:
+								skipLock: true
+						}, { force })
+					.then =>
+						@applications.executeStepAction(serviceAction('start', service.serviceId, null, service, skipLock: true), { force })
+					.then ->
+						res.status(200).json(Data: 'OK', Error: '')
+			.catch (err) ->
+				res.status(503).send(err?.message or err or 'Unknown error')
+		@router.use(@proxyvisor.router)
+
+module.exports = class ApplicationManager extends EventEmitter
+	constructor: ({ @logger, @config, @db, @eventTracker }) ->
+		@docker = new Docker()
+		@images = new Images({ @docker, @logger, @db })
+		@services = new ServiceManager({ @docker, @logger, @images, @config })
+		@networks = new Networks({ @docker, @logger })
+		@volumes = new Volumes({ @docker, @logger })
+		@proxyvisor = new Proxyvisor({ @config, @logger, @db, @docker, @images, applications: this })
+		@_targetVolatilePerServiceId = {}
+		@validActions = [
```

Comment:
This information is duplicated in the function names bellow. They should be in one place so that when the function name changes all references to it change too

Reply:
Done